File: src/ShoppingCart.Domain/Entities/Review.cs
using ShoppingCart.Domain.Entities.Base;

namespace ShoppingCart.Domain.Entities;

public class Review : AuditableEntity<int>
{
    public int ProductId { get; set; }
    public int CustomerId { get; set; }
    public int OrderId { get; set; }
    public int Rating { get; set; }
    public string? Title { get; set; }
    public string? Comment { get; set; }
    public bool IsVerifiedPurchase { get; set; } = false;
    public bool IsApproved { get; set; } = false;
    public bool IsFeatured { get; set; } = false;
    public int HelpfulCount { get; set; } = 0;
    public int NotHelpfulCount { get; set; } = 0;
    public string? AdminResponse { get; set; }
    public DateTime? AdminResponseDate { get; set; }

    public virtual Product Product { get; set; } = null!;
    public virtual Customer Customer { get; set; } = null!;
    public virtual Order Order { get; set; } = null!;
    public virtual ICollection<ReviewHelpfulness> HelpfulVotes { get; set; } = new List<ReviewHelpfulness>();
}

public class ReviewHelpfulness : AuditableEntity<int>
{
    public int ReviewId { get; set; }
    public int UserId { get; set; }
    public bool IsHelpful { get; set; }

    public virtual Review Review { get; set; } = null!;
}

File: frontend/shoppingcart-app/src/types/review.ts
export interface Review {
  id: number;
  productId: number;
  productName: string;
  productImageUrl?: string;
  customerId: number;
  customerName: string;
  customerAvatar?: string;
  rating: number;
  title?: string;
  comment?: string;
  isVerifiedPurchase: boolean;
  createdAt: string;
  adminResponse?: string;
  adminResponseDate?: string;
  helpfulCount: number;
  notHelpfulCount: number;
}

export interface ProductReviewSummary {
  productId: number;
  productName: string;
  averageRating: number;
  totalReviews: number;
  fiveStarCount: number;
  fourStarCount: number;
  threeStarCount: number;
  twoStarCount: number;
  oneStarCount: number;
  verifiedReviewsCount: number;
}

export interface CreateReviewRequest {
  productId: number;
  orderId: number;
  rating: number;
  title?: string;
  comment?: string;
}

export interface UpdateReviewRequest {
  id: number;
  rating: number;
  title?: string;
  comment?: string;
}

export interface ReviewListItem {
  id: number;
  productId: number;
  productName: string;
  customerName: string;
  rating: number;
  title?: string;
  comment?: string;
  isVerifiedPurchase: boolean;
  isApproved: boolean;
  isFeatured: boolean;
  createdAt: string;
  adminResponse?: string;
  adminResponseDate?: string;
}
